Formula/onedrive.rb

```diff
@@ -0,0 +1,33 @@
+class Onedrive < Formula
+  desc "Folder synchronization with OneDrive"
+  homepage "https://github.com/abraunegg/onedrive"
+  url "https://github.com/abraunegg/onedrive/archive/v2.4.12.tar.gz"
+  sha256 "4f6aa46fc28e859b63c0a2c190f66c3286ec0bde3f54f77af0ddea6b62bba37a"
+  license "GPL-3.0-only"
+
+  depends_on "dmd" => :build
+  depends_on "pkg-config" => :build
+  depends_on "curl"
+  depends_on :linux
+  depends_on "sqlite"
+
+  on_linux do
+    patch do
+      url "https://github.com/abraunegg/onedrive/commit/761cf3eb878fd370576329127055eec06b975672.patch?full_index=1"
+      sha256 "9a892e5e6bed7695754d49f9561d59f22180db7340493ec7f167bee767b8c28e"
+    end
+  end
```

Comment:
> be maintained (i.e. the last release wasn’t ages ago, it works without patching on all supported macOS releases and has no outstanding, unpatched security vulnerabilities)

https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff

This also counts for linux-only formulae, we really don't want to start this formulae off with a requirement for patches.

Reply:
Current release (`2.4.12`) does not compile against `dmd` v2.097 as mentioned in the commit(abraunegg/onedrive@761cf3eb878fd370576329127055eec06b975672). The patch block could be removed at the next release.

Reply:
Nevermind, it does not require the patch when using `ldc` instead `dmd` as compiler. I'll update the formula.